.github/workflows/pr.yml

```diff
@@ -0,0 +1,104 @@
+      - master
+  pull_request:
+env:
+  REALM_CI: true
+
+jobs:
+  baas-linux:
+    runs-on: ubuntu-latest
+    name: BaaS Linux
+    outputs:
+      clusterName: ${{ steps.deploy-mdb-apps.outputs.clusterName }}
+    timeout-minutes: 20
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+      - uses: realm/ci-actions/mdb-realm/deployApps@fac1d6958f03d71de743305ce3ab27594efbe7b7
+        id: deploy-mdb-apps
+        with:
+          projectId: ${{ secrets.ATLAS_PROJECT_ID }}
+          realmUrl: https://realm-dev.mongodb.com
+          atlasUrl: https://cloud-dev.mongodb.com
+          apiKey: ${{ secrets.ATLAS_PUBLIC_API_KEY }}
+          privateApiKey: ${{ secrets.ATLAS_PRIVATE_API_KEY }}
+          differentiator: dart-linux
+      - name : Setup Dart SDK
+        uses: dart-lang/setup-dart@main
+        with:
+          sdk: stable
+      - name: Deploy Apps
+        run: dart run realm_dart deploy-apps --baas-url https://realm-dev.mongodb.com --atlas-cluster ${{ steps.deploy-mdb-apps.outputs.clusterName }} --api-key ${{ secrets.ATLAS_PUBLIC_API_KEY }} --private-api-key ${{ secrets.ATLAS_PRIVATE_API_KEY }} --project-id ${{ secrets.ATLAS_PROJECT_ID }}
+  build-linux:
+    runs-on: ubuntu-latest
+    name: Build Linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Setup Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
+      - name: Build Realm Dart for Linux
+        run: ./scripts/build-linux.sh
+      - name: Store artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: librealm-linux
+          path: binary/**
+          retention-days: 1
+
+  tests-linux:
+    runs-on: ubuntu-latest
+    name: Tests Linux
+    env:
+      BAAS_URL: https://realm-dev.mongodb.com
+      BAAS_CLUSTER: ${{ needs.baas-linux.outputs.clusterName }}
+      BAAS_API_KEY: ${{ secrets.ATLAS_PUBLIC_API_KEY }}
+      BAAS_PRIVATE_API_KEY: ${{ secrets.ATLAS_PRIVATE_API_KEY }}
+      BAAS_PROJECT_ID: ${{ secrets.ATLAS_PROJECT_ID}}
+    needs:
+      - baas-linux
+      - build-linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Fetch artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: librealm-linux
+          path: binary
+      - name : Setup Dart SDK
+        uses: dart-lang/setup-dart@main
+        with:
+          sdk: stable
+      - name: Install dependencies
+        run: dart pub get
+      - name: Run tests
+        run: |
+          dart test -r expanded -j 1 --test-randomize-ordering-seed random
+
+  cleanup-baas:
+    runs-on: ubuntu-latest
+    name: Cleanup Linux
+    needs:
+      - tests-linux
+    if: always()
+    timeout-minutes: 5
+    steps:
+      - uses: realm/ci-actions/mdb-realm/cleanup@fac1d6958f03d71de743305ce3ab27594efbe7b7
```

Comment:
Could we tag this commit and reference it by tag?

Reply:
This is a bad practice because tags can be overwritten. It probably doesn't matter that much for our internal repositories, but I've always had it as a policy to use the commit sha for actions. [Here's the relevant section](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) from the Github docs.

Reply:
I see your point, but it makes it very hard to see if we are using the latest action version. Do you suggest we do the same for external actions?

Reply:
Ideally, this should be done for all actions, but at the very least, we should do it for actions where we pass secrets.